### ISSUE TITLE
FEATURE REQUEST: Delete/move local and remote

### ISSUE THREAD

**Issue by Fabrazer** [liximomo/vscode-sftp#321]
## Expected Behavior
In reference to following issue: #262 

All local file (and folder) operations in VS Code should also be performed remotely.

> I want to keep remote file operations in Remote Explorer.

For what reason? In my oppinion this only increases the workload.

## Actual Behavior
Files are often renamed, moved or deleted. All these operations are not performed remotely, which leads to unnecessary files on the remote server.

## Steps to Reproduce the Problem

  1. Create a file locally
  2. Rename or delete this file
  3. Take a look on the remote server -> the initial file can still be found on the remote server

## Specifications

  - Version: 1.5.7
  - Platform: Windows 10 64bit

**Comment by liximomo** [comment]
You can remove unwanted files from remote with `SFTP: Sync to Remote`. This will make sure the remote is exactly the same with local.

**Comment by Fabrazer** [comment]
Yeah, that's right. But that is another manual step that I would like to avoid. It could also be forgotten in everyday working life. Therefore, it is important for me that this happens automatically with the local operation.

Is there a way to trigger `SFTP: Sync to Remote` automatically together with the update / delete oparations? This would help for the moment.

Thank you!

**Comment by gsabater** [comment]
+1 to this,
I think it would be ideal to have a dropdown menu in the Right menu, and there we could select: remove local and remote, rename local and remote, etc.

**Comment by liximomo** [comment]
@Fabrazer @gergokoos rename and remove will come soon.

**Comment by gsabater** [comment]
Experimenting more on this, sync local -> remote lets you select a folder, but i couldn't find how to select a subfolder. I wanted to sync /root/foo, and it synced all /root with all their subfolders in a very big project.

In addition to add options like 
`rename local and remote file/folder`, `remove local and remote file/folder`, etc, i think that the `sync local -> remote` options should be done for the file or level you have opened / selected in the sidebar.

**Comment by liximomo** [comment]
> The primary situation in which this is super useful is if you're pulling a branch or checking out another branch, remote behaves accordingly.

How does `delete local and remote` work in this situation? It a branch checking out result a file get removed from local. How could you do a `delete local  and remote`.

**Comment by mstonepc** [comment]
> How does `delete local and remote` work in this situation? It a branch checking out result a file get removed from local. How could you run `delete local and remote` on an removed file?

You wouldn't.. you would just sync the local changes to remote. The watcher should be able to handle this scenario.

**Comment by CWrecker** [comment]
i need this

**Comment by liximomo** [comment]
@mstonepc 
```
"watcher": {
  files: "**/*",
  "autoUpload": true,
  "autoDelete": false
}
```

**Comment by Dshosev** [comment]
I would be glad to see an option for this behavior.
By this I mean that if I want the local actions be performed remotely, I can set the config for this.
In most of my projects the sync local > remote will take way too long.
But the user should decide what behavior is the right for his environment.

**Comment by b0nl** [comment]
Is there any indication when this would be released?
